File: app_pojav_zh/src/main/java/com/movtery/pojavzh/utils/file/FileUtils.java
package com.movtery.pojavzh.utils.file;

import java.io.InputStream;
import java.security.MessageDigest;

public class FileUtils {
    public static String getFileHashMD5(InputStream inputStream) {
        try {
            MessageDigest md = MessageDigest.getInstance("MD5");
            byte[] buffer = new byte[8192];
            int bytesRead;
            while ((bytesRead = inputStream.read(buffer)) != -1) {
                md.update(buffer, 0, bytesRead);
            }
            byte[] hash = md.digest();

            //将哈希值转换为十六进制字符串
            StringBuilder hexString = new StringBuilder();
            for (byte b : hash) {
                String hex = Integer.toHexString(0xff & b);
                if (hex.length() == 1) hexString.append('0');
                hexString.append(hex);
            }

            return hexString.toString();
        } catch (Exception e) {
            System.out.println(e.toString());
            return null;
        }
    }

    public static String getFileHashSHA1(InputStream inputStream) {
        try {
            MessageDigest md = MessageDigest.getInstance("SHA-1");
            byte[] buffer = new byte[8192];
            int bytesRead;
            while ((bytesRead = inputStream.read(buffer)) != -1) {
                md.update(buffer, 0, bytesRead);
            }
            byte[] hash = md.digest();

            //将哈希值转换为十六进制字符串
            StringBuilder hexString = new StringBuilder();
            for (byte b : hash) {
                String hex = String.format("%02x", b);
                hexString.append(hex);
            }

            return hexString.toString();
        } catch (Exception e) {
            System.out.println(e.toString());
            return null;
        }
    }
}

File: app_pojav_zh/src/main/java/com/movtery/pojavzh/utils/file/FileUtils.kt
package com.movtery.pojavzh.utils.file

import java.io.InputStream
import java.security.MessageDigest

object FileUtils {
    fun getFileHashMD5(inputStream: InputStream): String? {
        try {
            val md = MessageDigest.getInstance("MD5")
            val buffer = ByteArray(8192)
            var bytesRead: Int
            while ((inputStream.read(buffer).also { bytesRead = it }) != -1) {
                md.update(buffer, 0, bytesRead)
            }
            val hash = md.digest()

            //将哈希值转换为十六进制字符串
            val hexString = StringBuilder()
            for (b in hash) {
                val hex = Integer.toHexString(0xff and b.toInt())
                if (hex.length == 1) hexString.append('0')
                hexString.append(hex)
            }

            return hexString.toString()
        } catch (e: Exception) {
            return null
        }
    }

    @JvmStatic
    fun getFileHashSHA1(inputStream: InputStream): String? {
        try {
            val md = MessageDigest.getInstance("SHA-1")
            val buffer = ByteArray(8192)
            var bytesRead: Int
            while ((inputStream.read(buffer).also { bytesRead = it }) != -1) {
                md.update(buffer, 0, bytesRead)
            }
            val hash = md.digest()

            //将哈希值转换为十六进制字符串
            val hexString = StringBuilder()
            for (b in hash) {
                val hex = String.format("%02x", b)
                hexString.append(hex)
            }

            return hexString.toString()
        } catch (e: Exception) {
            return null
        }
    }
}
